webpack/webpack-env.d.ts

```diff
@@ -101,3 +102,137 @@ declare var __non_webpack_require__: any;
  * Equals the config option debug
  */
 declare var DEBUG: boolean;
+
+/**
+ * Webpack Hot Module Replacement
+ * https://webpack.github.io/docs/hot-module-replacement.html
+ */
+declare namespace __WebpackHotModuleReplacement {
+  interface Module {
```

Comment:
The copy-paste from `node.d.ts` is horrible, but I guess there is no way to import just what we want without the side-effects :( There's also a matter of what Webpack actually emulates from the node environment and that is controlled at runtime through the "node" option. I'm not sure which properties are actually supported. That should be checked. Accounting for the various configurations could be done with a union type, but that wouldn't be very practical as it would force users to explicitly cast `module` on every use. Any thoughts on that?

A few minor/cosmetic remarks here:
- I would move the namespace after the `__WebpackModuleApi` declaration, to keep the vars separate from the namespace definitions
- The `Module` stuff is better in `__WebpackModuleApi`. We might need to refactor it in the future if we were to split this file to support other targets (node etc.)